lib/ansible/modules/cloud/amazon/aws_waf_condition.py

```diff
@@ -551,11 +551,25 @@ def find_and_delete_condition(self, condition_set_id):
             self.tidy_up_regex_patterns(current_condition)
         return True, {}
 
+
+    def find_missing(self, update, current_condition):
+        missing = []
+        for desired in update['Updates']:
+            found = False
+            desired_condition = desired[self.conditiontuple]
+            current_conditions = current_condition[self.conditiontuples]
+            for current_condition in current_conditions:
+                if not compare_policies(current_condition, desired_condition):
+                    found = True
+            if not found:
+                missing.append(desired)
+        return missing
+
+
     def find_and_update_condition(self, condition_set_id):
         current_condition = self.get_condition_by_id(condition_set_id)
         update = self.format_for_update(condition_set_id)
-        missing = [desired for desired in update['Updates']
```

Comment:
This was checking if something like `{'FieldToMatch': {'Type': 'HEADER', 'Data': 'content-type'}, 'TextTransformation': 'NONE', 'TargetString': 'Hello', 'PositionalConstraint': 'STARTS_WITH'}` was in `{'FieldToMatch': {'Type': 'HEADER', 'Data': 'content-type'}, 'TargetString': b'Hello', 'TextTransformation': 'NONE', 'PositionalConstraint': 'STARTS_WITH'}` which would fail because b'Hello' != 'Hello'.